### PR TITLE
Binary Search in TypeScript

### LINKED ISSUE #2175: Binary Search
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [x] Dart - @keshakaneria
- [x] Go
- [x] Java
- [x] Julia
- [x] JS
- [x] Kotlin
- [x] Php
- [x] Python
- [x] Readme
- [x] Ruby
- [x] Rust - @raksha009
- [x] TypeScript - @Shr03mink

### PR DESCRIPTION
**Pull request by Shr03mink**
Fixes #2175 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-  TypeScript implementation of Binary Search algorithm
- 

#### Languages Used:
-  TypeScript
- 

#### Files Added:
-  binary_search.ts
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Shr03mink** [comment]
I tried add dynamic input using prompt but it was giving errors.

File: Binary_Search/binary_search.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/*    Implenetation of Binary Search algorithm to find value 6 in a given array of numbers    */
+
+function BinarySearch( arr:number[] , n:number ,  val:number ) :number
+{
+    var start = 0 //starting index of the array
+	var end = n-1 //ending index
```
Comment (raghav-dalmia) [inline Binary_Search/binary_search.ts:6]: improper indentation

File: Binary_Search/binary_search.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	var ind = -1  
+
+	while (start <= end)
+	{
+	    mid = Math.floor((start + end) / 2) 
+		
+		//checking if value to be found is at the middle 
+		if ( val == arr[mid] )
+		{
+            ind = mid
```
Comment (raghav-dalmia) [inline Binary_Search/binary_search.ts:17]: same here

File: Binary_Search/binary_search.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    console.log(" value ", val," found at index no. ", result," in the array ");
+}
+else {
+    console.log(" value ", val," not found ");
+}
+
+
+/*  OUTPUT:
+
+    value to be found:  6
```
Comment (raghav-dalmia) [inline Binary_Search/binary_search.ts:57]: same here

**Comment by raghav-dalmia** [comment]
Use 4 space indentation in your code

**Comment by Shr03mink** [comment]
I have made the requested changes and squashed the commits. Please let me know if any other change is required

**Pull request merged**